scripts/create-api.py

```diff
@@ -33,36 +36,55 @@ def linkify(text):
     return text
 
 
-def read_annotation(filename):
+def read_annotation(filename, valid_labels):
     annotation_md = frontmatter.load(filename)
 
     annotation = {"content": linkify(annotation_md.content)}
     for key in ["component", "features", "warning"]:
         if annotation_md.get(key):
             annotation[key] = annotation_md[key]
 
+    labels = annotation_md.get("labels")
+    if labels:
+        invalid_labels = [label for label in labels if label not in valid_labels]
+        if invalid_labels:
+            sys.stderr.write(
+                f"Invalid labels found in {annotation_filename}: {invalid_labels}; "
+                f" if these should be accepted values, update annotations/{app}/metadata.yaml"
+            )
+            sys.exit(1)
+        annotation["labels"] = labels
+
     return annotation
 
 
 for app in apps:
     try:
         data[app]["app"] = read_annotation(
-            os.path.join(ANNOTATIONS_DIR, app, "README.md")
+            os.path.join(ANNOTATIONS_DIR, app, "README.md"), []
```

Comment:
I updated the code in my scraper script to use Pathlib as @jklukas suggested, but modifying this one to do the same seems out of scope. Let's handle that in a followup.